sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -48,6 +48,15 @@ case class DataSourceV2Relation(
 
   import DataSourceV2Implicits._
 
+  override lazy val metadataOutput: Seq[AttributeReference] = table match {
+    case hasMeta: SupportsMetadataColumns =>
+      val attrs = hasMeta.metadataColumns
+      val outputNames = outputSet.map(_.name).toSet
+      attrs.filterNot(col => outputNames.contains(col.name)).toAttributes
```

Comment:
Can this possibly be buggy because of case-sensitivity playing a trick when filtering by Set.contains?

```suggestion
      val outputNames = outputSet.map(_.name.toLowerCase()).toSet
      attrs.filterNot(col => outputNames.contains(col.name.toLowerCase())).toAttributes
```